src/test/java/org/openmrs/contrib/qaframework/automation/EncountersSteps.java

```diff
@@ -0,0 +1,101 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.AdmitToInpatientPage;
+import org.openmrs.reference.page.PatientVisitsDashboardPage;
+import org.openmrs.uitestframework.test.TestData;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class EncountersSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+
+	@Before(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void visitHomePage() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Find Patient app from Home page")
+	public void goToFindPatientRecordPage() {
+		findPatientPage = homePage.goToFindPatientRecord();
+	}
+
+	@Then("the system loads find patient record page")
+	public void systemloadsFindPatientRecordPage() {
+		assertTrue(textExists("Find Patient Record"));
+	}
+
+	@When("a user searches an existing patient and starts the visit")
+	public void searchExistingPatient() {
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		visitsDashboardPage = findPatientPage.clickOnFirstPatient()
+				.startVisit();
```

Comment:
Can this be on same line.

Reply:
Any reason of inclining who functionalities at the same time.  **Search For exisiting Patient   + starting a visit**

Reply:
After simulating the test when start visit has been made to be start alone from clickOnFirstPatient() function, the test fails when it can not locate startVist() function even when its visible on the clinician dashboard page . Adding `waitForPage()` or `waitForPageToLoad()` does not make the function to be visible either. Investigation indicates that the issues are originating from the way how the page was designed in [distro](https://github.com/openmrs/openmrs-distro-referenceapplication/tree/master/ui-tests/src/test/java/org/openmrs/reference/page) and making a refactoring to suit the requirement will fail many selenium legacy test by default. 

Yes start Visit can be visible for the test to pass when the test logic has been refactored to pass via the Active visit but this wing does not prevail the encounters that are under test and makes the suggested wing to override the alternative.

Reply:
> Can this be on the same line 

We are glad that BDD framework workflow builds `auto-formats` the code base when they are being run locally. It would look very nice if we had some thing like `visitsDashboardPage.goToTransferToWardServicePage().confirm("Isolation Ward");` on the same line than having two lines at `73` and `74` but this is as result of auto formatting  that is than during the build plan.

Putting  it on the same line would make our code base look well I would say however, contributors working on new workflows will find a challenge when committing their changes since during the local build plan the workflow will auto-format all the existing test files and they will by default be tracked as changed lines introduced and this might not be a good move to trend.

When you look at other existing definition steps have this kind of format though it does not look good to me as well ! Example [FindPatientSteps](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a612c2ea4a56af898074d3227f026eff8010749/src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java#L29) [ConditionsSteps](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a612c2ea4a56af898074d3227f026eff8010749/src/test/java/org/openmrs/contrib/qaframework/automation/ConditionsSteps.java#L36) [ClinicalVisitSteps](https://github.com/openmrs/openmrs-contrib-qaframework/blob/6a612c2ea4a56af898074d3227f026eff8010749/src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java#L81)

Reply:
> the test fails when it can not locate startVist() function even when its visible on the clinician dashboard page

This is amazing work  .Why is the test failing to start a visit when other start visit functionalities have been working perfect, Am honestly not understanding this workflow. Do you mind sharing us where it fails we add a debugger point please.

Reply:
> but this wing does not prevail the encounters that are under test and makes the suggested wing to override the alternative.

What do you mean by wing

Reply:
> the test fails when it can not locate startVist() function even when its visible on the clinician dashboard page
> Why is the test failing to start a visit when other start visit functionalities have been working perfect

After fixing the issue which was originating from the server, the test logic as suggested by @sherrif10 worked liked  a charm !